### ISSUE TITLE
Unhelpful error message and/or type problem for `take`

### ISSUE THREAD

**Issue by zellyn** [robpike/ivy#41]
Perhaps I'm reading the docs wrong, but `ceil` is supposed to result in an integer; should this work?

```
n = 1 drop iota 10

(ceil (rho n)/2)
5

5 take n
2 3 4 5 6

(ceil (rho n)/2) take n
bad count for take
```

**Comment by robpike** [comment]
Simple bug in ceil (and perhaps some others). Fix coming.

**Comment by robpike** [comment]
Fixed in the dev branch by 5a2065c4984e4e4accfdb4868e132f0e273cc298
